install/setup_postgres.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+# The contents of this file are subject to the Common Public Attribution
+# License Version 1.0. (the "License"); you may not use this file except in
+# compliance with the License. You may obtain a copy of the License at
+# http://code.reddit.com/LICENSE. The License is based on the Mozilla Public
+# License Version 1.1, but Sections 14 and 15 have been added to cover use of
+# software over a computer network and provide for limited attribution for the
+# Original Developer. In addition, Exhibit A has been modified to be consistent
+# with Exhibit B.
+#
+# Software distributed under the License is distributed on an "AS IS" basis,
+# WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for
+# the specific language governing rights and limitations under the License.
+#
+# The Original Code is reddit.
+#
+# The Original Developer is the Initial Developer.  The Initial Developer of
+# the Original Code is reddit Inc.
+#
+# All portions of the code written by reddit are Copyright (c) 2006-2015 reddit
+# Inc. All Rights Reserved.
+###############################################################################
+set -x -e
+
+REDDIT_SRC="$1"
+echo $REDDIT_SRC
+
+if [ ! -e $REDDIT_SRC ]; then
+    echo "Couldn't find source '$REDDIT_SRC'. Aborting"
+    exit 1
+fi
+
+FUNCTIONSQL="$REDDIT_SRC/sql/functions.sql"
+
+if [ ! -e $FUNCTIONSQL ]; then
+    echo "Couldn't find functions.sql '$FUNCTIONSQL'. Aborting"
+    exit 1
+fi
+
+###############################################################################
+# Configure PostgreSQL
+###############################################################################
+SQL="SELECT COUNT(1) FROM pg_catalog.pg_database WHERE datname = 'reddit';"
+IS_DATABASE_CREATED=$(sudo -u postgres psql -t -c "$SQL")
+
+if [ $IS_DATABASE_CREATED -ne 1 ]; then
+    cat <<PGSCRIPT | sudo -u postgres psql
+CREATE DATABASE reddit WITH ENCODING = 'utf8' TEMPLATE template0 LC_COLLATE='en_US.utf8' LC_CTYPE='en_US.utf8';
+CREATE USER reddit WITH PASSWORD 'password';
+PGSCRIPT
+fi
+
+sudo -u postgres psql reddit <  $FUNCTIONSQL
```

Comment:
Don't feel strongly, but you could move all the stuff in functions.sql into this file to avoid needing to pass the path. It doesn't look like anything else is relying on functions.sql.